homeassistant/components/device_tracker/config_entry.py

```diff
@@ -165,23 +165,32 @@ def handle_device_event(ev: Event[EventDeviceRegistryUpdatedData]) -> None:
     hass.bus.async_listen(dr.EVENT_DEVICE_REGISTRY_UPDATED, handle_device_event)
 
 
-class BaseTrackerEntity(Entity):
+CACHED_PROPERTIES_WITH_ATTR_ = {
+    "battery_level",
+}
+
+
+class BaseTrackerEntity(Entity, cached_properties=CACHED_PROPERTIES_WITH_ATTR_):
     """Represent a tracked device."""
 
+    _attr_battery_level: int | None = None
     _attr_device_info: None = None
     _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_source_type: SourceType
 
     @cached_property
     def battery_level(self) -> int | None:
         """Return the battery level of the device.
 
         Percentage from 0-100.
         """
-        return None
+        return self._attr_battery_level
```

Comment:
We have an approved architecture discussion to deprecate this property, so I think we shouldn't add any new features around it.

https://github.com/home-assistant/architecture/discussions/627

Reply:
I have reverted the changes related to `battery_level`.